Add "Except DM" build mode to build.js

JJRcop opened the pull request:
## About The Pull Request
This adds a new build mode to tools/build/build, which runs everything except DM.

This might seem pretty useless, but my use case is I dev in a Linux env, and test my code in a Windows virtual machine. I would like to build the tgfont and tgui before transferring the files to the windows VM, and then finally run build there again, which will skip the tgfont and tgui steps because they were already completed.

**Further additions:**
902cdc840b de-duplicates the common portion of the tasksToRun array in each build mode.

JJRcop committed:
Add "Except DM" build mode to build.js

Mothblocks commented:
> This adds a new build mode to tools/build/build, which runs everything except CBT.

Do you mean except DM?

JJRcop commented:
> Do you mean except DM?

Well, CBT is a wrapper of DM, sorry for the confusion. I'll edit the description

JJRcop commented:
I forgot one of the more important things about case, that you must break each and every time

stylemistake commented:
Good addition and all that, but have you considered hosting the server on your Linux host, and then connect to it via Windows?

There is a huge advantage in chaining build with dream daemon command, and server startup is much faster on linux as well.

```
tools/build/build && DreamDaemon tgstation.dmb -trusted
```

JJRcop commented:
> have you considered hosting the server on your Linux host, and then connect to it via Windows?

I have considered this, yes. I might still do it, I mainly just don't want to install BYOND on my linux distro at the moment, but it's not that big of a deal.

Cyberboss commented on tools/build/build.js:
```diff
@@ -226,6 +230,13 @@ switch (BUILD_MODE) {
     break;
+  case NO_DM_BUILD:
+    tasksToRun = [
+      taskYarn,
+      taskTgfont,
+      taskTgui
```
This sequence is at the start of every build mode. Can we de-duplicate it?

JJRcop replied:
Sure

JJRcop committed:
Remove redundancy from build modes in tools/build/build.js

Replaced `^\s+tasksToRun = \[\n\s+taskYarn,\n\s+taskTgfont,\n\s+taskTgui,\n\s+([^\)]+\)),?\n\s+\];?`
with `    tasksToRun.push(\1);`
